paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/DataTableSource.java

```diff
@@ -191,22 +195,54 @@ public ScanRuntimeProvider getScanRuntimeProvider(ScanContext scanContext) {
             }
         }
 
+        List<Split> splits = null;
+        DataTableScan.DataFilePlan dataFilePlan = table.newScan().plan();
+        // TODO: we will refactor this after https://github.com/apache/incubator-paimon/issues/638
+        // finished.
+        if (null != dataFilePlan) {
+            splits = dataFilePlan.splits();
```

Comment:
There is also a cost to generating splits at the compile stage, it causes the generated topology to become large if there are a lot of splits.
Can we just generate splits when necessary and keep the original process without inference?

Reply:
@JingsongLi I update it